src/Adapter/CPDF.php

```diff
@@ -786,6 +786,39 @@ public function add_link($url, $x, $y, $width, $height)
         }
     }
 
+    public function font_supports_text(string $font, string $text)
```

Comment:
I did some quick testing using some (poorly) mocked up logic over 10000 iterations. For each test I checked to see if the codepoint existed in an array. The runs were:
1. generate a character code at random and run the check
2. pick a random character code from an array, put it in an array with a singular item, run the check in a foreach
3. pick a random character from an array, put it in a string, mb_strsplit => array_unique => array_map(mb_ord), run the check in a foreach

The last test attempting to be much like what's going on with this change. You can definitely [see the performance impact](https://3v4l.org/k6gbU#v8.2.5), though the time taken is still small enough that maybe it's OK. Still I'll see if I can work out some further improvement to the logic.

Reply:
In my mind this is less about optimization than about keeping it simple while solving the problem at hand. The current solution has a higher code complexity _and_ it incurs a performance cost, I think that is not worth it when all it gains is a maybe slightly more convenient interface for a hypothetical user; the current variant could always be implemented in addition, if there is demand or the need for it later. I see the `Canvas` interface as primarily serving the rendering process, and only secondarily as a convenient interface for manipulating the PDF directly – most users will never touch it directly, or only use the most basic methods.

That said, maybe it is possible to move some code around so that the current signature would make more sense.

Reply:
I testing variations of the method where the accepted second argument is one of the following:
- a string
- a single character
- a single character code

and did not see significant performance difference on my own system. I'll have to take a second look, but maybe because the FontMetrics method has already split the string up the overhead in the Canvas method is minimal (even though it's doing some of the same work).

I'll have to take another look to confirm but it seems like we're unlikely to squeeze much more performance out of this logic.

Reply:
Going to leave as-is for now. We can re-visit the optimization at a later date.